submit-web/src/components/Submission/VersionGroup.tsx

```diff
@@ -0,0 +1,57 @@
+import { useGetPackageVersions } from "@/hooks/api/usePackages";
+import { Backdrop, Button, CircularProgress } from "@mui/material";
+import { useParams, useRouter } from "@tanstack/react-router";
+import { useState } from "react";
+
+export default function VersionGroup({
+  packageId,
+  isPackageUpdating,
+  updatePackageId,
+}: {
+  packageId: number;
+  isPackageUpdating: boolean;
+  updatePackageId: (newPackageId: number) => void;
+}) {
+  const { projectId: accountProjectIdParam } = useParams({ strict: false });
+  const router = useRouter();
+  const [updatingPackageVersion, setUpdatingPackageVersion] = useState(false);
+
+  const { data: packageVersions } = useGetPackageVersions({
+    packageId: packageId,
+    enabled: Boolean(packageId),
+  });
+
+  function handleUpdatePackageId(newPackageId: number) {
+    setUpdatingPackageVersion(true);
+    updatePackageId(newPackageId);
+    router.navigate({
```

Comment:
why are we using router.navigate instead of navigate from navigate = useNavigate()